k2/csrc/ragged_ops.h

```diff
@@ -82,6 +81,26 @@ void LogSumPerSublist(Ragged<T> &src, T initial_value, Array1<T> *dst_values) {
   ApplyOpPerSublist<T, LogAdd<T>>(src, initial_value, dst_values);
 }
 
+
+/*
+  Output to an array `max_values` the arg-max within each sub-list along the
+  last axis of `src` i.e. the max taken over the last axis), i.e. the index
+  within `src.values` of the maximum element of that sub-list, or -1
+  if the sub-list was empty or had value <= `initial_value`.  [or maybe
```

Comment:
I think it should be `or had value < initial_value`. The implementation prefers the latter one if there is a tie.